File: app/src/main/java/example/com/shivangigotawalatest/Adapter/LanguageAdapter.java
package example.com.shivangigotawalatest.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import example.com.shivangigotawalatest.R;


public class LanguageAdapter extends RecyclerView.Adapter<LanguageAdapter.MyHolder> {

    private Context context;
    private List<String> languageList;

    public LanguageAdapter(Context context, List<String> languageList) {
        this.context = context;
        this.languageList = languageList;
    }

    @NonNull
    @Override
    public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.lang_info_ist_row, parent, false);
        return new LanguageAdapter.MyHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyHolder holder, int position) {
        StringBuffer tech = new StringBuffer(10);
        tech.append(languageList.get(position));
        tech.append("\n");
        holder.tvLang.setText(tech.toString());
    }

    @Override
    public int getItemCount() {
        return languageList.size();
    }

    public class MyHolder extends RecyclerView.ViewHolder {
        TextView tvLang;
        public MyHolder(View itemView) {
            super(itemView);
            tvLang = itemView.findViewById(R.id.tv_lang);
        }
    }
}

File: app/src/main/java/example/com/shivangigotawalatest/Adapter/LanguageAdapter.kt
package example.com.shivangigotawalatest.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import example.com.shivangigotawalatest.Adapter.LanguageAdapter.MyHolder

import example.com.shivangigotawalatest.R


class LanguageAdapter(private val context: Context, private val languageList: List<String>?) : RecyclerView.Adapter<MyHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.lang_info_ist_row, parent, false)
        return MyHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        val tech = StringBuffer(10)
        tech.append(languageList?.get(position))
        tech.append("\n")
        holder.tvLang!!.text = tech.toString()
    }

    override fun getItemCount(): Int {
        return languageList!!.size
    }

    inner class MyHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var tvLang: TextView? = null

        init {
            tvLang = itemView.findViewById(R.id.tv_lang)
        }
    }
}
